app/models/product.rb

```diff
@@ -15,6 +15,9 @@ class Product < ActiveRecord::Base
   has_many   :price_policies
   has_many   :training_requests, dependent: :destroy
 
+  cattr_accessor(:types) { ConfigurableArray.new([Instrument, Item, Service, Bundle]) }
+  cattr_accessor(:mergeable_types) { ConfigurableArray.new([Instrument, Item, Service, Bundle]) }
```

Comment:
The old hierarchy also exposes `Item.types`, so switching to `cattr_accessor` doesn't really change that behavior. In addition to typing, I do think it makes it a little easier to recognize as a setting.

I'm not opposed to moving towards `product_types`, but I think that's a big enough change to be outside the scope of this PR.